sphinx_immaterial/apidoc/format_signatures.py

```diff
@@ -79,167 +118,489 @@ def apply(self, **kwargs: Any) -> None:
-            node = convert_desc_parameterlist(node)
-        if isinstance(node, docutils.nodes.TextElement):
-            return process_text_element(node)
-        raise ValueError("unexpected child")
-
-    node["is_multiline"] = False
-    for child in children:
-        _extend_children_copy_source_info(node, child, process(child))
-
-    if name_text_node_replacement is not None:
-        (
-            existing_node,
-            new_node,
-        ) = name_text_node_replacement
-        source = existing_node.source
-        line = existing_node.line
-        existing_node.parent.replace(existing_node, new_node)
-        new_node.source = source
-        new_node.line = line
+                cur_replacements = replacements[id(components[component_i].node)]
+                if tag == "equal":
+                    value = components[component_i].orig_text[
+                        i1 - component_start_offset : i2 - component_start_offset
+                    ]
+                    i1 += len(value)
+                    cur_replacements.append(SignatureText(value))
+                else:
+                    while j1 < j2:
+                        if (
+                            m := _PUNCTUATION_PATTERN.search(formatted_output, j1, j2)
+                        ) is not None:
+                            m_start = m.start()
+                            m_end = m.end()
+                        else:
+                            m_start = m_end = j2
+                        if j1 != m_start:
+                            cur_replacements.append(
+                                SignatureText(formatted_output[j1:m_start])
+                            )
+                        if m_start != m_end:
+                            t = formatted_output[m_start:m_end]
+                            cur_replacements.append(
+                                sphinx.addnodes.desc_sig_punctuation(t, t)
+                            )
+                        j1 = m_end
+                    i1 = min(i2, component_end_offset)
+                if i1 == i2:
+                    break
+
+        del signature_to_modify[:]
+        signature_to_modify.extend(
+            _replace_text_nodes(self.adjusted_nodes, self.ignored_nodes, replacements)
+        )
+
+        if "sig-wrap" in signature_to_modify["classes"]:
+            signature_to_modify["classes"].remove("sig-wrap")
+
+        signature_to_modify["is_multiline"] = False
+
+
+def _format_signature_with_black(
+    domain: str, objtype: str, node: sphinx.addnodes.desc_signature, options
+):
+    import black
+
+    black_format_style = options.get("black_format_style")
+    assert black_format_style is not None
+
+    mode = black.FileMode(
+        line_length=black_format_style.line_length
+        or options["wrap_signatures_column_limit"],
+        string_normalization=black_format_style.string_normalization,
+    )
+
+    applyer = FormatApplier()
```

Comment:
I still get spelling errors for this variable.
```suggestion
    applier = FormatApplier()
```

FYI, I'm using a [vscode extension to check spelling](https://github.com/streetsidesoftware/vscode-spell-checker). We could add a [codespell](https://github.com/codespell-project/codespell) check to CI, but it would have a lot of problems with the member names from sphinx/docutils.

Reply:
Done.